src/imitation/util/logger.py

```diff
@@ -0,0 +1,197 @@
+
+    Args:
+      subdir: A string key which determines the `folder` where raw data is
+        written and temporary logging prefixes for raw and mean data. Entering
+        an `accumulate_means` context in the future with the same `subdir`
+        will safely append to logs written in this folder rather than
+        overwrite.
+    """
+    if self.current_logger is not None:
+      raise RuntimeError("Nested `accumulate_means` context")
+
+    if subdir in self._cached_loggers:
+      logger = self._cached_loggers[subdir]
+    else:
+      folder = os.path.join(self.default_logger.dir, "raw", subdir)
+      os.makedirs(folder)
+      output_formats = _build_output_formats(folder, self.format_strs)
+      logger = sb_logger.Logger(folder, output_formats)
+      self._cached_loggers[subdir] = logger
+
+    try:
+      self.current_logger = logger
+      self._subdir = subdir
+      yield
+    finally:
+      self.current_logger = None
+      self._subdir = None
+
+  def logkv(self, key, val):
+    if self.current_logger is not None:
+      assert self._subdir is not None
+      raw_key = os.path.join("raw", self._subdir, key)
+      self.current_logger.logkv(raw_key, val)
+
+      mean_key = os.path.join("mean", self._subdir, key)
+      self.default_logger.logkv_mean(mean_key, val)
+    else:
+      self.default_logger.logkv_mean(key, val)
+
+  @property
+  def _logger(self):
+    if self.current_logger is not None:
+      return self.current_logger
+    else:
+      return self.default_logger
+
+  def logkv_mean(self, key, val):
+    self._logger.logkv_mean(key, val)
+
+  def dumpkvs(self):
+    self._logger.dumpkvs()
+
+  def log(self, *args, **kwargs):
+    self._logger.log(*args, **kwargs)
+
+  def get_dir(self) -> str:
+    return self._logger.get_dir()
+
+  def close(self):
+    self._logger.close(self)
+
+
+def _sb_logger_configure_replacement(*args, **kwargs):
+  raise RuntimeError("Shouldn't call stable_baselines.logger.configure "
+                     "once imitation.logger.configure() has been called")
+
+
+def _sb_logger_reset_replacement():
+  raise RuntimeError("Shouldn't call stable_baselines.logger.reset "
+                     "once imitation.logger.configure() has been called")
+
+
+def is_configured() -> bool:
+  """Return True if the custom logger is active."""
+  return isinstance(sb_logger.Logger.CURRENT, _HierarchicalLogger)
+
+
+def configure(folder: str, format_strs: Optional[Sequence[str]] = None) -> None:
+  """Configure Stable Baselines logger to be `accumulate_means()`-compatible.
+
+  After this function is called, `stable_baselines.logger.{configure,reset}()`
+  are replaced with stubs that raise RuntimeError.
+
+  Args:
+      folder: Argument from `stable_baselines.logger.configure`.
+      format_strs: An list of output format strings. For details on available
+        output formats see `stable_baselines.logger.make_output_format`.
+  """
+  sb_logger.configure = _sb_logger_configure_replacement
```

Comment:
Not clear what problem this is preventing. Add comment maybe?

Reply:
This is just a sanity check, don't want a gotcha from someone calling `stable_baselines.logger.configure` after `imitation.util.logger.configure`